modules/activemq-artemis-launch/added/drain.sh

```diff
@@ -1,177 +1,244 @@
-  fi
-done <<< ${dnsNames}
-IFS=$IFSP
 
-if [ -z "$hostName" ]; then
-  echo "[drain.sh] Can't find target host name"
-  exit 1
-fi
+    SCALE_TO_BROKER="${hostName}"
+    log "scale down target is: $SCALE_TO_BROKER"
 
-source /opt/amq/bin/launch.sh nostart
+    # Add connector to the pod to scale down to
+    log "removing any existing scaledownconnector"
+    sed -i "s/<connector name=\"scaledownconnector\">.*//" ${instanceDir}/etc/broker.xml
 
-SCALE_TO_BROKER="${hostName}"
-echo "[drain.sh] scale down target is: $SCALE_TO_BROKER"
+    log "adding new connector"
+    connector="<connector name=\"scaledownconnector\">tcp:\/\/${SCALE_TO_BROKER}:61616<\/connector>"
+    sed -i "/<\/connectors>/ s/.*/${connector}\n&/" ${instanceDir}/etc/broker.xml
 
-# Add connector to the pod to scale down to
-connector="<connector name=\"scaledownconnector\">tcp:\/\/${SCALE_TO_BROKER}:61616<\/connector>"
-sed -i "/<\/connectors>/ s/.*/${connector}\n&/" ${instanceDir}/etc/broker.xml
+    # Remove the acceptors
+    acceptor="<acceptor name=\"artemis\">tcp:\/\/${BROKER_HOST}:61616?protocols=CORE<\/acceptor>"
+    sed -i -ne "/<acceptors>/ {p; i $acceptor" -e ":a; n; /<\/acceptors>/ {p; b}; ba}; p" ${instanceDir}/etc/broker.xml
 
-# Remove the acceptors
-#sed -i -ne "/<acceptors>/ {p;   " -e ":a; n; /<\/acceptors>/ {p; b}; ba}; p" ${instanceDir}/etc/broker.xml
-acceptor="<acceptor name=\"artemis\">tcp:\/\/${BROKER_HOST}:61616?protocols=CORE<\/acceptor>"
-sed -i -ne "/<acceptors>/ {p; i $acceptor" -e ":a; n; /<\/acceptors>/ {p; b}; ba}; p" ${instanceDir}/etc/broker.xml
+    # start the broker and issue the scaledown command to drain the messages.
+    log "launch the drainer broker"
 
-#start the broker and issue the scaledown command to drain the messages.
-${instanceDir}/bin/artemis-service start
+    ${instanceDir}/bin/artemis-service start
 
-tail -n 100 -f ${AMQ_NAME}/log/artemis.log &
+    #no longer log to file by default
+    #tail -n 100 -f ${AMQ_NAME}/log/artemis.log &
 
-waitForJolokia
+    waitForJolokia
 
-RET_CODE=`curl -G -k http://${AMQ_USER}:${AMQ_PASSWORD}@${BROKER_HOST}:8161/console/jolokia/exec/org.apache.activemq.artemis:broker=%22${AMQ_NAME}%22/scaleDown/scaledownconnector`
+    # calculate total messages
+    get_total_messages_on_broker
+    total_before_scaledown=$TOTAL_MESSAGES_ON_BROKER
 
-HTTP_CODE=`echo $RET_CODE | python2 -c "import sys, json; print json.load(sys.stdin)['status']"`
+    log "initiating scaledown. There are $total_before_scaledown messages to be migrated"
+    mm_start=$(date +%s)
 
-echo "[drain.sh] curl return code ${HTTP_CODE}"
+    RET_CODE=`curl -s -G -k http://${AMQ_USER}:${AMQ_PASSWORD}@${BROKER_HOST}:8161/console/jolokia/exec/org.apache.activemq.artemis:broker=%22${AMQ_NAME}%22/scaleDown/scaledownconnector`
 
-if [ "${HTTP_CODE}" != "200" ]; then
-  echo "[drain.sh] scaleDown is not successful, response: $RET_CODE"
-  echo "[drain.sh] sleeping for 30 seconds to allow inspection before it restarts"
-  sleep 30
```

Comment:
the previous script sleeps for 30 seconds after a failure to allow inspection, does the new script allows inspection?

Reply:
The sleep is moved to when all the pods have tried and failed. here https://github.com/artemiscloud/activemq-artemis-broker-kubernetes-image/pull/81/files#diff-16b237f518d394fb048aa263553f35a607e7d6ae8ecf9bb9b2cb6a73a78c7a7fR97